temporalio/client.py

```diff
@@ -4734,3 +4922,247 @@ def _fix_history_enum(prefix: str, parent: Dict[str, Any], *attrs: str) -> None:
+                promote_build_id_within_set=self.build_id
+            )
+        )
+
+
+@dataclass
+class BuildIdOpMergeSets(BuildIdOperation):
+    """Merges two sets into one set, thus declaring all the Build Ids in both as compatible with one
+    another. The default of the primary set is maintained as the merged set's overall default.
+    """
+
+    primary_build_id: str
+    """A Build Id which and is used to find the primary set to be merged."""
+
+    secondary_build_id: str
+    """A Build Id which and is used to find the secondary set to be merged."""
+
+    def as_partial_proto(
+        self,
+    ) -> temporalio.api.workflowservice.v1.UpdateWorkerBuildIdCompatibilityRequest:
+        return temporalio.api.workflowservice.v1.UpdateWorkerBuildIdCompatibilityRequest(
+            merge_sets=temporalio.api.workflowservice.v1.UpdateWorkerBuildIdCompatibilityRequest.MergeSets(
+                primary_set_build_id=self.primary_build_id,
+                secondary_set_build_id=self.secondary_build_id,
+            )
+        )
+
+
+@dataclass
+class WorkerTaskReachability:
+    """Contains information about the reachability of some Build IDs"""
+
+    build_id_reachability: Mapping[str, BuildIdReachability]
+    """Maps Build IDs to information about their reachability"""
+
+    @staticmethod
+    def _from_proto(
+        resp: temporalio.api.workflowservice.v1.GetWorkerTaskReachabilityResponse,
+    ) -> WorkerTaskReachability:
+        mapping = dict()
+        for bid_reach in resp.build_id_reachability:
+            tq_mapping = dict()
+            unretrieved = set()
+            for tq_reach in bid_reach.task_queue_reachability:
+                if tq_reach.reachability == [
+                    temporalio.api.enums.v1.TaskReachability.TASK_REACHABILITY_UNSPECIFIED
+                ]:
+                    unretrieved.add(tq_reach.task_queue)
+                    continue
+                tq_mapping[tq_reach.task_queue] = [
+                    ReachabilityType._from_proto(r) for r in tq_reach.reachability
+                ]
+
+            mapping[bid_reach.build_id] = BuildIdReachability(
+                task_queue_reachability=tq_mapping,
+                unretrieved_task_queues=frozenset(unretrieved),
+            )
+
+        return WorkerTaskReachability(build_id_reachability=mapping)
+
+
+@dataclass
+class BuildIdReachability:
+    """Contains information about the reachability of a specific Build ID"""
+
+    task_queue_reachability: Mapping[str, Sequence[ReachabilityType]]
+    """Maps Task Queue names to the reachability status of the Build ID on that queue. If the value
+    is an empty list, the Build ID is not reachable on that queue.
+    """
+
+    unretrieved_task_queues: FrozenSet[str]
+    """If any Task Queues could not be retrieved because the server limits the number that can be 
+    queried at once, they will be listed here.
+    """
+
+
+class ReachabilityType(IntEnum):
+    NEW_WORKFLOWS = 1
```

Comment:
With some of the other enums we were just putting the proto values as the enum values, though I don't particularly mind not doing it here, just mentioning.

Reply:
Note, might as well not even make it a `IntEnum` if there's no value to the integers